src/http/service/web/service.rs

```diff
@@ -145,7 +145,7 @@ where
     }
 
     /// add a route to the web service which matches the given matcher, using the given service.
-    pub fn on<I, T>(mut self, matcher: HttpMatcher, service: I) -> Self
+    pub fn on<I, T>(mut self, matcher: HttpMatcher<State, Body>, service: I) -> Self
```

Comment:
The `Body` type parameter is `http::Body`, the default type for a `Request`.

Reply:
Yes that's okay for the high level web service, agreed